virtual-host-gatherer/lib/gatherer/modules/Xen.py

```diff
@@ -0,0 +1,169 @@
+# Copyright (c) 2021 SUSE LLC, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Xen Worker module implementation.
+"""
+
+from __future__ import print_function, absolute_import
+import logging
+from collections import OrderedDict
+from gatherer.modules import WorkerInterface
+
+try:
+    import XenAPI
+    IS_VALID = True
+except ImportError:
+    IS_VALID = False
+
+
+
+class Xen(WorkerInterface):
+    """
+    Worker class for Xen.
+    """
+
+    DEFAULT_PARAMETERS = OrderedDict([
+        ('hostname', ''),
+        ('port', '80'),
+        # ('port', '443'),
+        ('username', ''),
+        ('password', '')])
+
+    VMSTATE = {
+        'Halted': 'stopped',
+        'Running': 'running',
+        'Paused': 'paused',
+        'Suspended': 'suspended',
+    }
+
+    def __init__(self):
+        """
+        Constructor.
+
+        :return:
+        """
+
+        self.log = logging.getLogger(__name__)
+        self.host = self.port = self.user = self.password = None
+
+    # pylint: disable=R0801
+    def set_node(self, node):
+        """
+        Set node information
+
+        :param node: Dictionary of the node description.
+        :return: void
+        """
+
+        try:
+            self._validate_parameters(node)
+        except AttributeError as error:
+            self.log.error(error)
+            raise error
+
+        self.host = node['hostname']
+        self.port = node.get('port', 80)
+        self.user = node['username']
+        self.password = node['password']
+
+    def parameters(self):
+        """
+        Return default parameters
+
+        :return: default parameter dictionary
+        """
+
+        return self.DEFAULT_PARAMETERS
+
+    def run(self):
+        """
+        Start worker.
+
+        :return: Dictionary of the hosts in the worker scope.
+        """
+        output = dict()
+        self.log.info('Connect Xen Server %s:%s as user %s', self.host, self.port, self.user)
+        try:
+            session = XenAPI.Session("http://%s:%s" % (self.host, self.port))
```

Comment:
One question: regardless of the port you specify, shouldn't we use `https` protocol here?

Reply:
I don't think so. If the user has configured an insecure transport (HTTP over port 80), that's what we need to use, otherwise it would fail. I don't think anyone would configure HTTPS on port 80 ¿?